lib/mqtt/mqtt.cc

```diff
@@ -583,13 +633,12 @@ SObj mqtt_connect(Timeout                    *t,
 	// Prepare the low-level MQTT API connect call.
 	MQTTConnectInfo_t connectInfo = {0};
 
-	// We want to create a new session with the broker.
-	connectInfo.cleanSession = true;
-
+	connectInfo.cleanSession           = newSession;
 	connectInfo.pClientIdentifier      = clientID;
```

Comment:
Why not capture this as a std::string_view?  Does it need to remain valid across calls?  If so, we should document that.  If not, then we need to document that it does need to have global because, even though we're not capturing it, we are stashing it in a heap object.  Would it make sense to just copy this, since it's only 23 bytes?

Reply:
The client ID is only needed for the connect. It does not need to be valid across calls. `connectInfo` is not a heap object, it's on the stack?

I suggest to leave things as they are and just document the lifetime of the client ID in the API.